libminifi/include/utils/MinifiConcurrentQueue.h

```diff
@@ -21,14 +21,49 @@
 #include <deque>
 #include <mutex>
 #include <condition_variable>
+#include <utility>
 #include <stdexcept>
+#include <type_traits>
 
 namespace org {
 namespace apache {
 namespace nifi {
 namespace minifi {
 namespace utils {
 
+namespace detail {
```

Comment:
This is good stuff, but I think it should be in GeneralUtils.h as this is not a concurrentQueue-specific functionality.

Reply:
+1 for extracting the `void_t` implementation but -1 for `TryMoveCall` (at least until we need it elsewhere).
If `void_t` is extracted, please adapt the partial detection idiom impl above `StringUtils::join_pack` to use the extracted one.

Reply:
Extracted both functions to their according header, updated `StringUtils` to include `void_t` from the external definition.